traits/trait_numeric.py

```diff
@@ -351,13 +365,29 @@ class CArray(AbstractArray):
         second dimension must be at least 2.)
     value : numpy array
         A default value for the array.
+    casting : str
+        Casting rule for ``numpy.ndarray.astype``. Values that cannot be
```

Comment:
Can we reword this docstring a bit to say what the effect of this keyword is, so that the reader doesn't have to go digging into the NumPy documentation to figure it out?

It may also be worth clarifying that this parameter is not used if `dtype` is not given.

Reply:
I was thinking of something along the lines of:

> Rule used to convert array values to the target dtype. The default is `"unsafe"`, which
> allows arbitrary data conversions. `"safe"` permits only conversions that don't change
> the value (so for example conversion from `float64` to `float32` would not be permitted).
> The meaning and behaviour are exactly as in ``numpy.ndarray.astype``. see the NumPy
> documentation for other options for this parameter.

Reply:
For bonus points, we could even make this a proper Sphinx link and add NumPy to the intersphinx configuration.